strawberryfields/apps/data/sample.py

```diff
@@ -372,3 +372,111 @@ class Formic(MoleculeDataset):
     n_mean = 1.56
     threshold = False
     T = 0
+
+
+class Water(SampleDataset):
+    """Vibrational dynamics of the water molecule.
+
+    The molecular parameters are obtained from Ref. :cite:`sparrow2018simulating`.
+
+    **Molecule:**
+
+    .. |water| image:: ../../../_static/water.png
+        :align: middle
+        :width: 200px
+        :target: javascript:void(0);
+
+    |water|
+
+    Attributes:
+        n_mean = 1/3
+        threshold = False
+        n_samples = 135000
+        modes = 3
+    """
+
+    times_to_indices = {t: i for i, t in enumerate(np.linspace(0, 270, 28))}
+
+    # pylint: disable=super-init-not-called
+    def __init__(self, t):
+        if t not in self.times_to_indices:
+            raise ValueError(
+                "The selected time is not correct. Available times are 0, 10, 20, " "..., 270"
+            )
+        index = self.times_to_indices[t]
+
+        all_data = np.load(DATA_PATH + "water.npz")["arr_0"]
+
+        self.data = all_data[index]
+        self.data = scipy.sparse.csr_matrix(self.data)
+        self.n_samples, self.modes = self.data.shape
+
+    w = scipy.sparse.load_npz(DATA_PATH + "water_w.npz").toarray()[0]
+    U = scipy.sparse.load_npz(DATA_PATH + "water_U.npz").toarray()
+
+    n_mean = 1 / 3
+    threshold = False
+    _data_filename = "water"
+
+    def available_times(self):
+        """Returns the times for which pre-generated samples are available."""
+        return list(self.times_to_indices.keys())
+
+
+class Pyrrole(SampleDataset):
+    """Vibrational dynamics of the pyrrole molecule.
+
+    The molecular parameters are obtained from Ref. :cite:`jahangiri2020quantum`.
+
+    **Molecule:**
+
+    .. |pyrrole| image:: ../../../_static/pyrrole.png
+        :align: middle
+        :width: 350px
+        :target: javascript:void(0);
+
+    |pyrrole|
+
+    Attributes:
+        n_mean = 0.12599583
+        threshold = False
+        n_samples = 10000
+        modes = 24
+    """
+
+    times_to_indices = {t: i for i, t in enumerate(np.linspace(0, 1000, 11))}
+
+    # pylint: disable=super-init-not-called
+    def __init__(self, t):
+        if t not in self.times_to_indices:
+            raise ValueError(
+                "The selected time is not correct. Available times are 0, 100, 200, " "..., 1000"
+            )
+        index = self.times_to_indices[t]
+
+        all_data = np.load(DATA_PATH + "pyrrole.npz")["arr_0"]
+
+        self.data = all_data[index]
+        self.data = scipy.sparse.csr_matrix(self.data)
+        self.n_samples, self.modes = self.data.shape
+
+    ri = scipy.sparse.load_npz(DATA_PATH + "pyrrole_ri.npz").toarray()[0]

```

Comment:
Could these arrays be saved all in one `pyrrole_data.npz` file? Here we don't need to be using sparse arrays.